state/factory/statedb.go

```diff
@@ -213,7 +213,11 @@ func (sdb *stateDB) Height() (uint64, error) {
 }
 
 func (sdb *stateDB) newWorkingSet(ctx context.Context, height uint64) (*workingSet, error) {
-	flusher, err := db.NewKVStoreFlusher(sdb.dao, batch.NewCachedBatch(), sdb.flusherOptions(ctx, height)...)
+	var opts []batch.CachedBatchOption
+	if genesis.MustExtractGenesisContext(ctx).IsMidway(height) {
+		opts = append(opts, batch.SafeRevertOption())
+	}
+	flusher, err := db.NewKVStoreFlusher(sdb.dao, batch.NewCachedBatch(opts...), sdb.flusherOptions(ctx, height)...)
```

Comment:
after 1.7.0, we can verify sync and revert this

Reply:
add TODO